ports/esp32/usb_serial_jtag.c

```diff
@@ -33,11 +33,38 @@
 #include "hal/usb_serial_jtag_ll.h"
 #include "esp_intr_alloc.h"
 #include "soc/periph_defs.h"
+#include "freertos/portmacro.h"
 
 #define USB_SERIAL_JTAG_BUF_SIZE (64)
 
 static uint8_t rx_buf[USB_SERIAL_JTAG_BUF_SIZE];
 static volatile bool terminal_connected = false;
+static DRAM_ATTR portMUX_TYPE rx_mux = portMUX_INITIALIZER_UNLOCKED;
+
+void usb_serial_jtag_handle_rx(void) {
+    if (xPortInIsrContext()) {
```

Comment:
Suggest doing an upfront check of `if (!usb_serial_jtag_ll_rxfifo_data_available()) return;` before anything else (including taking the spinlock), as most of the time there won't be anything for this function to do.

Reply:
I've instead added this to the new `usb_serial_jtag_poll_rx` function as per above comment, since if called from the ISR there this check is redundant.